.github/workflows/build-test-e2e.yml

```diff
@@ -1,11 +1,34 @@
 name: 'Build/Run End-to-End Tests'
 
 on: push
+env:
+  BUILD_ARTIFACT_NAME: swc-war
+  BUILD_DIR: target
+  BUILD_NAME: portal-develop-SNAPSHOT.war
+  NODE_VERSION: 18.16.0
+  PW_ALL_BLOBS_DIR: all-blob-reports
 jobs:
-  build-and-run-e2e:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Build SWC
+        uses: ./.github/workflows/build
+      - name: Upload build to GitHub Actions Artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ env.BUILD_ARTIFACT_NAME }}
+          path: '${{ env.BUILD_DIR }}/${{ env.BUILD_NAME }}'
+          retention-days: 1
```

Comment:
Build SWC once, then upload WAR as an artifact

Reply:
Specify the full file path, so that other files in the `target` directory are not included in the artifact

Reply:
This is unique per-workflow, right? i.e. if you have 2 PRs running this job, could you have this race condition? 

1. Job 1 builds + uploads WAR artifact 1
2. Job 2 builds + uploads WAR artifact 2
3. Job 1 shard(s) downloads the artifact, which ends up incorrectly being WAR artifact 2

Reply:
That's a good question! I think artifacts are unique per workflow run (e.g. [each e2e workflow run](https://github.com/Sage-Bionetworks/SynapseWebClient/actions/workflows/build-test-e2e.yml) has its own 'playwright-report' artifact), but I couldn't find something in the docs that explicitly states this. 

Either way, I don't think we'd run into that race condition because per the [GH docs](https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts#about-workflow-artifacts), "You can only download artifacts that were uploaded during the same workflow run." So the Job 1 shard should only be able to download WAR artifact 1, even if another WAR artifact exists.